src/Microsoft.AspNetCore.Razor.Language/Resources.resx

```diff
@@ -156,43 +156,43 @@
+  <data name="TagHelper_InvalidBoundAttributePrefix" xml:space="preserve">
+    <value>Invalid tag helper bound property '{1}' on tag helper '{0}'. Tag helpers cannot bind to HTML attributes with prefix '{2}' because the prefix contains a '{3}' character.</value>
   </data>
-  <data name="InvalidBoundAttributePrefixStartsWith" xml:space="preserve">
-    <value>Invalid tag helper bound property '{0}.{1}'. Tag helpers cannot bind to HTML attributes with prefix '{2}' because the prefix starts with '{3}'.</value>
+  <data name="TagHelper_InvalidBoundAttributePrefixStartsWith" xml:space="preserve">
+    <value>Invalid tag helper bound property '{1}' on tag helper '{0}'. Tag helpers cannot bind to HTML attributes with prefix '{2}' because the prefix starts with '{3}'.</value>
   </data>
-  <data name="InvalidRestrictedChild" xml:space="preserve">
-    <value>Invalid restricted child '{0}' for tag helper '{1}'. Tag helpers cannot restrict child elements that contain a '{2}' character.</value>
+  <data name="TagHelper_InvalidRestrictedChild" xml:space="preserve">
+    <value>Invalid restricted child '{1}' for tag helper '{0}'. Tag helpers cannot restrict child elements that contain a '{2}' character.</value>
   </data>
-  <data name="InvalidRestrictedChildNullOrWhitespace" xml:space="preserve">
+  <data name="TagHelper_InvalidRestrictedChildNullOrWhitespace" xml:space="preserve">
     <value>Invalid restricted child for tag helper '{0}'. Name cannot be null or whitespace.</value>
   </data>
-  <data name="InvalidTargetedAttributeName" xml:space="preserve">
+  <data name="TagHelper_InvalidTargetedAttributeName" xml:space="preserve">
     <value>Tag helpers cannot target attribute name '{0}' because it contains a '{1}' character.</value>
   </data>
-  <data name="InvalidTargetedAttributeNameNullOrWhitespace" xml:space="preserve">
+  <data name="TagHelper_InvalidTargetedAttributeNameNullOrWhitespace" xml:space="preserve">
     <value>Targeted attribute name cannot be null or whitespace.</value>
   </data>
-  <data name="InvalidTargetedParentTagName" xml:space="preserve">
+  <data name="TagHelper_InvalidTargetedParentTagName" xml:space="preserve">
     <value>Tag helpers cannot target parent tag name '{0}' because it contains a '{1}' character.</value>
   </data>
-  <data name="InvalidTargetedParentTagNameNullOrWhitespace" xml:space="preserve">
+  <data name="TagHelper_InvalidTargetedParentTagNameNullOrWhitespace" xml:space="preserve">
     <value>Targeted parent tag name cannot be null or whitespace.</value>
   </data>
-  <data name="InvalidTargetedTagName" xml:space="preserve">
+  <data name="TagHelper_InvalidTargetedTagName" xml:space="preserve">
     <value>Tag helpers cannot target tag name '{0}' because it contains a '{1}' character.</value>
   </data>
-  <data name="InvalidTargetedTagNameNullOrWhitespace" xml:space="preserve">
+  <data name="TagHelper_InvalidTargetedTagNameNullOrWhitespace" xml:space="preserve">
```

Comment:
There's a bunch more of these btw that don't include the tag helper name - my guess is because of the design it was too hard. My take on the builders makes it pretty easy for us to do this in the future.